Oeuvres listes

baudu69 opened the pull request.

baudu69 committed:
Liste des oeuvres back #2

AlexisGiust committed:
Liste des auteurs back #5

AlexisGiust committed:
Fix navbar CSS

AlexisGiust committed:
Premier jet liste des auteurs front #6

AlexisGiust committed:
Lazy load + log

baudu69 committed:
Liste des oeuvres front #3

AlexisGiust commented on pratibook-api/src/main/java/fr/inextenso/pratibook/model/Cree.java:
```diff
@@ -0,0 +1,32 @@
+
+@Entity
+@Table(name = "cree")
+public class Cree {
```
C'est pas plus propre d'utiliser un @ManyToMany sur les classes Auteur et Oeuvre ?

AlexisGiust commented on pratibook-api/src/main/java/fr/inextenso/pratibook/model/Disponibilite.java:
```diff
@@ -0,0 +1,28 @@
+
+    private final int value;
+
+    Disponibilite(int value) {
```
private ?

baudu69 replied:
Pourquoi pas ?

AlexisGiust commented on pratibook-api/src/main/java/fr/inextenso/pratibook/model/GenreOeuvre.java:
```diff
@@ -0,0 +1,19 @@
+
+@Entity
+@Table(name = "genre_oeuvre")
+public class GenreOeuvre {
```
Pareil que pour la classe Cree, @ManyToMany ?

AlexisGiust commented on pratibook-api/src/main/java/fr/inextenso/pratibook/service/ServiceOeuvre.java:
```diff
@@ -0,0 +1,75 @@
+                .toList();
+    }
+
+    private List<AuteurOeuvreDTO> getAuteurOeuvreDTOs(String idOeuvre) {
```
Simplifiable avec le @ManyToMany

AlexisGiust commented on pratibook-api/src/main/java/fr/inextenso/pratibook/service/ServiceOeuvre.java:
```diff
@@ -0,0 +1,75 @@
+                .toList();
+    }
+
+    private List<GenreOeuvreDTO> getGenreOeuvreDTOs(String idOeuvre) {
```
Pareil

AlexisGiust commented on pratibook-ui/src/app/liste-oeuvre/liste-oeuvre.service.ts:
```diff
@@ -0,0 +1,16 @@
+  constructor(private http: HttpClient) { }
+
+  getAllOeuvres(): Observable<HttpResponse<Oeuvre[]>> {
+    return this.http.get<Oeuvre[]>(`/api/oeuvre`, {observe: 'response'});
```
Pourquoi ` et pas ' ?

AlexisGiust committed:
Merge branch 'oeuvres-listes' into auteurs-liste

AlexisGiust committed:
Merge pull request #17 from baudu69/auteurs-liste

Liste des auteurs #4

baudu69 committed:
Ajout des listes d'auteurs et d'oeuvres dans la liste des requetes autorises de l'API. #4 #1
Liste d'auteur accessible a tout le monde #4
Filtre sur le nom et le prenom de l'auteur #1

baudu69 committed:
Filtre sur le genre de l'oeuvre #1
